### DISCUSSION TITLE
Button in Kotlin allows single word to split into two lines?  How to get single line text?

### DISCUSSION THREAD

**Discussion by davidakoontz** [skiptools/skip#214] [Q&A] [answered]
I'm trying to understand how to fix the Kotlin code that allows a single word text in the Button to split into two lines.  See the simulator images below.  The iPhone displays a reasonable button with 4 letter words.  The Android typically breaks the word into multiple lines - why?  How do I fix it?

Here is the Kotlin code (which I don't really know much about):
`                    ScrollView { ->
                        ComposeBuilder { composectx: ComposeContext ->
                            LazyVGrid(columns = columns, spacing = vgrid_vert_spacing) { ->
                                ComposeBuilder { composectx: ComposeContext ->
                                    ForEach(0..<words.count, id = { it }) { iter ->
                                        ComposeBuilder { composectx: ComposeContext ->
                                            Button({
                                                val str = LocalizedStringKey.StringInterpolation(literalCapacity = 0, interpolationCount = 0)
                                                str.appendInterpolation(words[iter])
                                                LocalizedStringKey(stringInterpolation = str)
                                            }()) { ->
                                                appState.alpha = words[iter]
                                                dismiss() // uses Env to go Back a Page
                                            }
                                            .buttonStyle(ButtonStyle.bordered)
                                            .tint(Color.green)
                                            .foregroundColor(Color.black).Compose(composectx)
                                            ComposeResult.ok
                                        }
                                    }.Compose(composectx)
                                    ComposeResult.ok
                                }
                            }`
<img width="564" alt="Screenshot 2024-08-28 at 12 55 19 PM" src="https://github.com/user-attachments/assets/5b3eb7e9-abc0-41a3-ad5d-8a3d6be07ede">

simulator (Android left - iPhone right)

![Screenshot 2024-08-28 at 1 02 00 PM](https://github.com/user-attachments/assets/16bd3bf1-dce8-46af-a56f-d9660db52d12)

**Comment by aabewhite** [comment, marked as answer]
What do your “columns” GridItems look like? Android seems much more sensitive to the given "minimum" than iOS

**Reply by davidakoontz** [reply]
Oh THANKS!  Yes!  I was looking to local... I don't yet know much about the GridItem.  But I had grid_h_min = 66.  Looked/worked in iOS.  Just changed to 80 and Kotlin quit splitting the text in the Button.  So that magic number works!

![Screenshot 2024-08-28 at 2 18 03 PM](https://github.com/user-attachments/assets/bbf3a978-31e1-4505-8c38-73e82e1cb2f2)

`  //Using GridItem(.adaptive(minimum: 80)) means we want the grid to fit in as many items per row as possible, using a minimum size of 80 points each.
    static let  grid_h_min = 80.0
    let columns = [
        GridItem(.adaptive(minimum: grid_h_min))
        //TODO: work on GridItem settings
    ]`

**Reply by davidakoontz** [reply]
that FIX makes me so happy...  I was thinking I was going to have to mutate the Kotlin files after Skip generated them... the nuclear option is never the way!
